optuna/storages/_heartbeat.py

```diff
@@ -68,6 +70,59 @@ def get_failed_trial_callback(self) -> Optional[Callable[["optuna.Study", Frozen
         raise NotImplementedError()
 
 
+class BaseHeartbeatThread:
+    def start(self, trial_id: int) -> None:
+        raise NotImplementedError()
+
+    def join(self) -> None:
+        raise NotImplementedError()
```

Comment:
[Question] This class has only abstract methods. Could you explain why you avoid `ABCMeta` unlike the `BaseStorage` and `BaseSampler`?

https://github.com/optuna/optuna/blob/07a8193c34e5e389770c6ae549001bb574510e7a/optuna/storages/_base.py#L22

Reply:
Right; this definitely can be an ABCMeta class. I changed there.

Reply:
Thanks!